### PR TITLE
create load

### PR DESCRIPTION
**Pull request by FranckLecuyer**
PR linked to another PR : https://github.com/gridsuite/study-server/pull/112

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 1]
```diff
@@ -34,43 +36,45 @@
     @Autowired
     private NetworkModificationService networkModificationService;
 
-    @PutMapping(value = "/networks/{networkUuid}/switches/{switchId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @PutMapping(value = "/networks/{networkUuid}/group/{groupUuid}/switches/{switchId}", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:39]: groups ?

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 2]
```diff
@@ -81,13 +85,23 @@
         return ResponseEntity.ok().body(networkModificationService.getModificationGroups());
     }
 
-    @PutMapping(value = "/networks/{networkUuid}/lines/{lineId}/status", consumes = MediaType.TEXT_PLAIN_VALUE)
+    @PutMapping(value = "/networks/{networkUuid}/group/{groupUuid}/lines/{lineId}/status", consumes = MediaType.TEXT_PLAIN_VALUE)
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:88]: API looks strange, groups is not a sub resource of network. 
Not easy to design but I think we should choose if the primary (I means most important) is network or modification group and then specify the the second infos with query param.
Like /networks/{networkUuid}/lines/{lineId}/status&group={groupUuid}.
Maybe in some use case the group is not necessary as we might (not for gridsuite) be able to modify the network without recording the modification.

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: Done

File: src/main/java/org/gridsuite/modification/server/dto/EquipmentCreationInfos.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+@SuperBuilder
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString(callSuper = true)
+@Schema(description = "Equipment creation")
+public class EquipmentCreationInfos extends ElementaryModificationInfos {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/dto/EquipmentCreationInfos.java:25]: it is not an elementary modification of equipment ?

File: src/main/java/org/gridsuite/modification/server/entities/elementary/CreateEquipmentEntity.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    private String equipmentId;
+
+    @Column(name = "equipmentName")
+    private String equipmentName;
+
+    @Column(name = "voltageLevelId")
+    private String voltageLevelId;
+
+    @Column(name = "busId")
+    private String busId;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/entities/elementary/CreateEquipmentEntity.java:36]: NodeBreaker topology ?

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java [hunk 5]
```diff
@@ -99,13 +102,22 @@ public void saveModifications(UUID groupUuid, List<ModificationEntity> modificat
...
                 .filter(m -> groupUuid.equals(m.getGroup().getId()))
                 .orElseThrow(() -> new NetworkModificationException(MODIFICATION_NOT_FOUND, modificationUuid.toString())))
-                .toElementaryModificationInfos();
+                .toElementaryAttributeModificationInfos();
+    }
+
+    public LoadCreationInfos getLoadCreationModification(UUID groupUuid, UUID modificationUuid) {
+        return ((CreateLoadEntity) this.modificationRepository
+            .findById(modificationUuid)
+            .filter(m -> ModificationType.LOAD_CREATION.name().equals(m.getType()))
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:117]: Type et groupUUId not necessary ?

File: src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java [hunk 6]
```diff
@@ -21,56 +24,98 @@
...
-    private UUID networkUuid;
+    private UUID groupUuid;
 
     private Network network;
 
     private final NetworkModificationRepository modificationRepository;
 
     private List<ElementaryModificationEntity<?>> modifications = new LinkedList<>();
 
+    private List<CreateEquipmentEntity> creations = new LinkedList<>();
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java:35]: Is it really necessary to separate the creations
Yhe listener saves all modifications regardless of the type

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 7]
```diff
@@ -197,6 +204,18 @@ private boolean disconnectLineBothSides(Network network, String lineId) {
         }
     }
 
+    private List<ElementaryModificationInfos> doActionCreation(NetworkStoreListener listener, Runnable action, NetworkModificationException.Type typeIfError) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:207]: Why separate the creations from other modifications ?
The treatment is generic

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/entities/elementary/CreateEquipmentEntity.java (hunk 4 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/entities/elementary/CreateEquipmentEntity.java:36]: busId here is a generic term for BusBarSection id in node breaker topology, and Bus id in bus breaker topology

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 8]
```diff
@@ -37,40 +39,42 @@
...
+                                                                                        @RequestParam(value = "group", required = false) UUID groupUuid,
+                                                                                        @RequestBody String groovyScript) {
+        return ResponseEntity.ok().body(networkModificationService.applyGroovyScript(networkUuid, groupUuid, groovyScript));
     }
 
-    @GetMapping(value = "/networks/{networkUuid}/modifications", produces = MediaType.APPLICATION_JSON_VALUE)
-    @Operation(summary = "Get modifications list of the default group for a network")
-    @ApiResponse(responseCode = "200", description = "List of modifications of the default group for the network")
-    public ResponseEntity<Flux<ModificationInfos>> getModifications(@Parameter(description = "Group UUID") @PathVariable("networkUuid") UUID groupUuid) {
+    @GetMapping(value = "/networks/modifications/group/{groupUuid}", produces = MediaType.APPLICATION_JSON_VALUE)
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: Done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java [hunk 9]
```diff
@@ -24,7 +25,11 @@
         MODIFICATION_NOT_FOUND(HttpStatus.NOT_FOUND),
         SWITCH_NOT_FOUND(HttpStatus.NOT_FOUND),
         LINE_NOT_FOUND(HttpStatus.NOT_FOUND),
-        MODIFICATION_ERROR(HttpStatus.INTERNAL_SERVER_ERROR);
+        MODIFICATION_ERROR(HttpStatus.INTERNAL_SERVER_ERROR),
+        VOLTAGE_LEVEL_NOT_FOUND(HttpStatus.INTERNAL_SERVER_ERROR),
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:29]: NOT_FOUND ?

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java [hunk 10]
```diff
@@ -24,7 +25,11 @@
...
+        CREATE_LOAD_ERROR(HttpStatus.INTERNAL_SERVER_ERROR),
+        BUSBAR_SECTION_NOT_FOUND(HttpStatus.INTERNAL_SERVER_ERROR),
+        BUS_NOT_FOUND(HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:32]: Same

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java [hunk 11]
```diff
@@ -24,7 +25,11 @@
         MODIFICATION_NOT_FOUND(HttpStatus.NOT_FOUND),
         SWITCH_NOT_FOUND(HttpStatus.NOT_FOUND),
         LINE_NOT_FOUND(HttpStatus.NOT_FOUND),
-        MODIFICATION_ERROR(HttpStatus.INTERNAL_SERVER_ERROR);
+        MODIFICATION_ERROR(HttpStatus.INTERNAL_SERVER_ERROR),
+        VOLTAGE_LEVEL_NOT_FOUND(HttpStatus.INTERNAL_SERVER_ERROR),
+        CREATE_LOAD_ERROR(HttpStatus.INTERNAL_SERVER_ERROR),
+        BUSBAR_SECTION_NOT_FOUND(HttpStatus.INTERNAL_SERVER_ERROR),
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:31]: Same

Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:31]: Done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java (hunk 10 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:32]: Done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java (hunk 9 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:29]: Done

File: src/main/java/org/gridsuite/modification/server/entities/elementary/EquipmentCreationEntity.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
+/*
+  Copyright (c) 2021, RTE (http://www.rte-france.com)
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.modification.server.entities.elementary;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/entities/elementary/EquipmentCreationEntity.java:7]: Create new packages `org.gridsuite.modification.server.entities.equipment.creation `and `org.gridsuite.modification.server.entities.equipment.attribute.modification` ?

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 13]
```diff
@@ -172,25 +219,31 @@ private boolean disconnectLineBothSides(Network network, String lineId) {
...
     }
 
-    private List<ElementaryModificationInfos> doAction(Network network, UUID networkUuid, Runnable action, NetworkModificationException.Type typeIfError) {
+    private List<EquipmenModificationInfos> doAction(Network network, UUID groupUuid, Runnable modification, NetworkModificationException.Type typeIfError) {
+        NetworkStoreListener listener = NetworkStoreListener.create(network, groupUuid, modificationRepository);
+        ReporterModel reporter = new ReporterModel("NetworkModification", "Network modification");
+        return doAction(listener, modification, typeIfError, reporter);
+    }
+
+    private List<EquipmenModificationInfos> doAction(NetworkStoreListener listener, Runnable action, NetworkModificationException.Type typeIfError, Reporter reporter) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:236]:
```
    private List<EquipmenModificationInfos> doAction(NetworkStoreListener listener, Runnable action, NetworkModificationException.Type typeIfError, Reporter reporter) {
        try {
            action.run();
            saveModifications(listener);
            return listener.getModifications();
        } catch (Exception e) {
            NetworkModificationException exc = e instanceof NetworkModificationException ? (NetworkModificationException)e :  new NetworkModificationException(typeIfError, e);
            reporter.report(Report.builder()
                .withKey(typeIfError.name())
                .withDefaultMessage(exc.getMessage())
                .withSeverity(new TypedValue("NETWORK_MODIFICATION_ERROR", TypedValue.ERROR_LOGLEVEL))
                .build());
            throw exc;
        }
    }
```

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 14]
```diff
@@ -172,25 +219,31 @@ private boolean disconnectLineBothSides(Network network, String lineId) {
...
-    private List<ElementaryModificationInfos> doAction(Network network, UUID networkUuid, Runnable modification) {
-        return doAction(network, networkUuid, modification, MODIFICATION_ERROR);
+    private List<EquipmenModificationInfos> doAction(Network network, UUID groupUuid, Runnable modification) {
+        return doAction(network, groupUuid, modification, MODIFICATION_ERROR);
     }
 
-    private List<ElementaryModificationInfos> doAction(Network network, UUID networkUuid, Runnable action, NetworkModificationException.Type typeIfError) {
+    private List<EquipmenModificationInfos> doAction(Network network, UUID groupUuid, Runnable modification, NetworkModificationException.Type typeIfError) {
+        NetworkStoreListener listener = NetworkStoreListener.create(network, groupUuid, modificationRepository);
+        ReporterModel reporter = new ReporterModel("NetworkModification", "Network modification");
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:232]: The `ReporterModel` `NetworkModification` must be global for a study
It must be created when the study is created ?

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 13 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:236]: Do not forget to do the send report in a finaly clause

Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java]: Done

File: src/main/java/org/gridsuite/modification/server/entities/elementary/EquipmentCreationEntity.java (hunk 12 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/entities/elementary/EquipmentCreationEntity.java]: Done

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java (hunk 5 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:124]: Both methods getEquipmentAttributeModification and getLoadCreationModification are used only in tests
filtering with the type ensure the cast to the right entity
filtering with the groupUuid verifies that the modification is in the right group

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 14 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:232]: An another US

**Review by SlimaneAmar — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/dto/EquipmentType.java [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+
+import com.powsybl.iidm.network.*;
+import org.gridsuite.modification.server.NetworkModificationException;
+
+import java.util.Objects;
+
+/**
+ * @author Slimane Amar <slimane.amar at rte-france.com>
+ */
+public enum EquipmentType {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/dto/EquipmentType.java:17]: Unused?

File: src/main/java/org/gridsuite/modification/server/dto/EquipmentCreationInfos.java [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+@SuperBuilder
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString(callSuper = true)
+@Schema(description = "Equipment creation")
+public class EquipmentCreationInfos extends EquipmenModificationInfos {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/dto/EquipmentCreationInfos.java:25]: Only keep equipmentName and create a InjectionCreationInfos with voltageLevelId and busId

File: src/main/java/org/gridsuite/modification/server/entities/equipment/creation/EquipmentCreationEntity.java [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+import javax.persistence.Column;
+import javax.persistence.MappedSuperclass;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+@NoArgsConstructor
+@Getter
+@MappedSuperclass
+public class EquipmentCreationEntity extends EquipmentModificationEntity {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/entities/equipment/creation/EquipmentCreationEntity.java:23]: Same as for dto

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 18]
```diff
@@ -220,4 +277,123 @@ private void saveModifications(NetworkStoreListener listener) {
...
+            .setConnectableBus(bus.getId())
+            .setP0(loadCreationInfos.getActivePower())
+            .setQ0(loadCreationInfos.getReactivePower())
+            .add();
+    }
+
+    public Flux<EquipmenModificationInfos> createLoad(UUID networkUuid, UUID groupUuid, LoadCreationInfos loadCreationInfos) {
+        return assertLoadCreationInfosNotEmpty(loadCreationInfos).thenMany(
+                getNetwork(networkUuid).flatMapIterable(network -> {
+                    NetworkStoreListener listener = NetworkStoreListener.create(network, groupUuid, modificationRepository);
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:348]: I think you mix here 2 concepts: listener et modification list.
The storage is done synchronously with `listener.storeLoadCreation(loadCreationInfos)` but you still use the concept of `NetworkStoreListener` which is not implemented for creation.
Add a `NetworkModificationList` which can be used inside a `NetworkStoreListener` or standalone like here.

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 19]
```diff
@@ -220,4 +277,123 @@ private void saveModifications(NetworkStoreListener listener) {
...
+        BusbarSection busbarSection = nodeBreakerView.getBusbarSection(loadCreationInfos.getBusId());
+        if (busbarSection == null) {
+            throw new NetworkModificationException(BUSBAR_SECTION_NOT_FOUND, loadCreationInfos.getBusId());
+        }
+
+        // creating the disconnector
+        int newNode = nodeBreakerView.getMaximumNodeIndex();
+        nodeBreakerView.newSwitch()
+                .setId("disconnector_" + loadCreationInfos.getEquipmentId())
+                .setName("disconnector_" + loadCreationInfos.getEquipmentName())
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:293]: What about if `loadCreationInfos.getEquipmentName()` is not defined? I think we will end up with an ugly and useless name

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 20]
```diff
@@ -220,4 +277,123 @@ private void saveModifications(NetworkStoreListener listener) {
...
+                .setOpen(false)
+                .setFictitious(false)
+                .setNode1(busbarSection.getTerminal().getNodeBreakerView().getNode())
+                .setNode2(newNode + 1)
+                .add();
+
+        // creating the breaker
+        nodeBreakerView.newSwitch()
+            .setId("breaker_" + loadCreationInfos.getEquipmentId())
+            .setName("breaker_" + loadCreationInfos.getEquipmentName())
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:305]: Same

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 21]
```diff
@@ -220,4 +277,123 @@ private void saveModifications(NetworkStoreListener listener) {
...
+                    }, CREATE_LOAD_ERROR, networkUuid, reporter, subReporter);
+                }));
+    }
+
+    private Mono<Void> assertLoadCreationInfosNotEmpty(LoadCreationInfos loadCreationInfos) {
+        return loadCreationInfos == null ? Mono.error(new NetworkModificationException(CREATE_LOAD_ERROR, "Missing required attributes to create the load")) : Mono.empty();
+    }
+
+    private void sendReport(UUID networkUuid, ReporterModel reporter) {
+        AtomicReference<Long> startTime = new AtomicReference<>(System.nanoTime());
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:382]: What is the purpose of `AtomicReference ?`

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 22]
```diff
@@ -220,4 +277,123 @@ private void saveModifications(NetworkStoreListener listener) {
...
+        var headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        var resourceUrl = DELIMITER + REPORT_API_VERSION + DELIMITER + "reports" + DELIMITER + networkUuid.toString();
+        var uriBuilder = UriComponentsBuilder.fromPath(resourceUrl);
+        try {
+            reportServerRest.exchange(uriBuilder.toUriString(), HttpMethod.PUT, new HttpEntity<>(objectMapper.writeValueAsString(reporter), headers), ReporterModel.class);
+        } catch (JsonProcessingException error) {
+            throw new PowsyblException("error creating report", error);
+        } finally {
+            LOGGER.trace("Save reports for network '{}' in parallel : {} seconds", networkUuid, TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - startTime.get()));
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:392]: In parallel of what?

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 23]
```diff
@@ -37,44 +38,38 @@
...
-    public ResponseEntity<Mono<Void>> deleteModificationGroup(@Parameter(description = "Group UUID") @PathVariable("networkUuid") UUID groupUuid) {
+    @DeleteMapping(value = "/groups/{groupUuid}")
+    @Operation(summary = "Delete the modifications group for a network")
+    @ApiResponse(responseCode = "200", description = "Modifications group deleted for the network")
+    public ResponseEntity<Mono<Void>> deleteModificationGroup(@Parameter(description = "Group UUID") @PathVariable("groupUuid") UUID groupUuid) {
         return ResponseEntity.ok().body(networkModificationService.deleteModificationGroup(groupUuid));
     }
 
-    @GetMapping(value = "/networks/modificationgroups", produces = MediaType.APPLICATION_JSON_VALUE)
+    @GetMapping(value = "/modificationgroups", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:72]: /groups to be consistent with other methods

File: src/main/java/org/gridsuite/modification/server/dto/EquipmentType.java (hunk 15 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/dto/EquipmentType.java]: Removed

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 19 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java]: Done

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 20 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java]: Done

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 21 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java]: Removed

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java (hunk 23 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: Done

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 18 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:346]: Will be done in another US

**Pull request merged**